src/pal/pal_linux.h

```diff
@@ -7,6 +7,11 @@
 #  include <string.h>
 #  include <sys/mman.h>
 
+#  ifndef SNMALLOC_PLATFORM_HAS_GETENTROPY
```

Comment:
Do we actually need the CMake check?  Linux exposes the syscall numbers as macros, so we should be able to just include the header and check if it defines the syscall that we expect.  We could probably do this unconditionally for Linux, since I think any Linux version that is currently supported, including LTS kernels, will support this system call.